packages/webpack-plugin/src/loader-utils.ts

```diff
@@ -1,46 +1,86 @@
 import type { Stylable, StylableMeta } from '@stylable/core';
 import { tryCollectImportsDeep } from '@stylable/core/dist/index-internal';
-import { processUrlDependencies, hasImportedSideEffects } from '@stylable/build-tools';
+import {
+    processUrlDependencies,
+    collectSheetsWithSideEffects,
+    hasImportedSideEffects,
+} from '@stylable/build-tools';
 import { LOADER_NAME } from './plugin-utils';
+import { dirname, relative } from 'path';
 
 export function getReplacementToken(token: string) {
     return `/* INJECT */ {__${token}__:true}`;
 }
 
+export function normalizeRelative(p: string) {
+    p = p.replace(/\\/g, '/');
+    return p.startsWith('.') ? p : './' + p;
+}
+
 export function getImports(
     stylable: Stylable,
     meta: StylableMeta,
     projectRoot: string,
     assetFilter: (url: string, context: string) => boolean,
-    assetsMode: 'url' | 'loader'
+    assetsMode: 'url' | 'loader',
+    includeGlobalSideEffects: boolean
 ) {
     const urls = processUrlDependencies(meta, projectRoot, assetFilter);
     const imports: string[] = [];
     const unusedImports: string[] = [];
 
-    for (const imported of meta.getImportStatements()) {
-        // attempt to resolve the request through stylable resolveModule,
-        // is case of an error fall back to the original request
-        let resolved = imported.request;
-        try {
-            resolved = stylable.resolver.resolvePath(imported.context, imported.request);
-        } catch (e) {
-            // fallback to request
+    if (includeGlobalSideEffects) {
+        // new mode that collect deep side effects
+        const resolvedSideEffectRequests = collectSheetsWithSideEffects(stylable, meta);
+        const contextDir = dirname(meta.source);
+        // add side-effects sheets as imports
+        for (const resolvedRequest of resolvedSideEffectRequests) {
+            const relativePath = normalizeRelative(relative(contextDir, resolvedRequest));
```

Comment:
webpack and rollup don't need relative paths in the modules. make sure to remove the resolve in plugin.ts
for both modes!

Reply:
in the Cli it will be important to use the relative